synapse/event_auth.py

```diff
@@ -637,6 +671,66 @@ def get_user_power_level(user_id: str, auth_events: StateMap[EventBase]) -> int:
             return 0
 
 
+def get_users_which_can_issue_invite(auth_events: StateMap[EventBase]) -> List[str]:
```

Comment:
I'm a bit worried that this is returning a list of potential servers? It feels like a recipe for different implementations to do signature checks on different servers and thus potentially come to different conclusions (though it seems fairly unlikely).

Reply:
This is now only used to find a list of candidate servers to attempt a remote join to.

Reply:
I'd be tempted to lift these two functions out of here, as this file is otherwise stuff purely related to the auth rules check from the spec?